nasa-gibs/worldview
# WV-1979: Add event tracks as overlays

## Description

Changes event tracks to be rendered as overlays rather than vector layers.  This prevents cluttering the OL map layers collection with non-imagery layers, which helps avoid bugs where we make assumptions that this collection only includes imagery layers (which has come up a few times now).

* Clean up code, break out some functions to a util file
* Render track lines as an SVG polyline
* Move track point  date tooltips to show _above_ points so mouse cursor doesn't obscure
  * Tooltips still show below for selected point so even icon doesn't obscure
* Fix an issue where switching from Events to Layers tab during animation would cause event tracks/points to not be removed from the map

---

## How to Test

1. Load app with ice/storm events: http://localhost:3000/?e=true&efs=true&efd=2021-07-25,2021-11-22&efc=severeStorms,seaLakeIce&t=2021-11-22-T17%3A03%3A25Z
2. For each of the following, after taking an action confirm tracks/points render as expected 
3. Switch between events 
4. Switch between points on a track (clicking point directly) 
5. Switch between points on a track (clicking date in event list) 
6. Drag/zoom with tracks rendered 
---
1. Load app with event selected and animation set-up: [here](http://localhost:3000/?v=88.66484375000002,7.880859375,124.73515625000002,31.119140625&as=2021-10-14-T00%3A00%3A00Z&ae=2021-10-24-T00%3A00%3A00Z&e=EONET_5947,2021-10-14&efs=true&efd=2021-07-25,2021-11-22&efc=severeStorms,seaLakeIce&l=MODIS_Aqua_Sea_Ice(hidden),MODIS_Terra_Sea_Ice(hidden),VIIRS_SNPP_Brightness_Temp_BandI5_Night(hidden),VIIRS_SNPP_Brightness_Temp_BandI5_Day(hidden),VIIRS_NOAA20_Brightness_Temp_BandI5_Night(hidden),VIIRS_NOAA20_Brightness_Temp_BandI5_Day(hidden),MODIS_Aqua_Brightness_Temp_Band31_Night(hidden),MODIS_Aqua_Brightness_Temp_Band31_Day(hidden),MODIS_Terra_Brightness_Temp_Band31_Night(hidden),MODIS_Terra_Brightness_Temp_Band31_Day(hidden),IMERG_Precipitation_Rate,VIIRS_SNPP_DayNightBand_ENCC(hidden),VIIRS_SNPP_DayNightBand_At_Sensor_Radiance(hidden),VIIRS_SNPP_DayNightBand_AtSensor_M15(hidden),Reference_Labels_15m,Reference_Features_15m,Coastlines_15m(hidden),VIIRS_NOAA20_CorrectedReflectance_TrueColor(hidden),VIIRS_SNPP_CorrectedReflectance_TrueColor,MODIS_Aqua_CorrectedReflectance_TrueColor(hidden),MODIS_Terra_CorrectedReflectance_TrueColor&lg=true&ab=on&t=2021-10-14-T00%3A00%3A00Z)
2. Play animation
3. Switch back to layers tab
4. Confirm event tracks/points get removed


@nasa-gibs/worldview
